src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -836,6 +836,20 @@ get_softkernels(const axlf* top)
   return sks;
 }
 
+aie_partition_obj
+get_aie_partition(const axlf* top)
+{
+  auto pSection = ::xclbin::get_axlf_section(top, AIE_PARTITION);
+  if (!pSection)
+    return {};
+
+  auto begin = reinterpret_cast<const char*>(top) + pSection->m_sectionOffset;
+  auto aiep = reinterpret_cast<const aie_partition*>(begin);
+  auto scp = reinterpret_cast<const uint16_t*>(begin + aiep->info.mpo_auint16_start_columns);
+
+  return {aiep->info.column_width, {scp, scp + aiep->info.start_columns_count}, begin + aiep->mpo_name};
```

Comment:
trying to understand how "begin + aiep->mpo_name" is the name of partition? is mpo_name actually the offset to the actual string?

Reply:
Yes, anything start from mpo_ is an offset to the actual object. e.g. mpo_aunit16_start_columns